db/crud.go

```diff
@@ -1754,6 +1754,12 @@ func (db *Database) updateAndReturnDoc(docid string, allowImport bool, expiry ui
 				doc.inlineSyncData = true
 			}
 
+			userXattrs := currentUserXattr
+			if existingDoc != nil {
```

Comment:
If existingDoc is populated by checkForUpgrade, on line 1736 above, it doesn't look like UserXattr will be properly populated without a change to db.GetDocWithXattr, which I don't see here.

Reply:
Correct, I'd missed that case, good spot.

Reply:
Fixed as per changes to GetWithXattr discussed